tests/utils/test_shell_util.py

```diff
@@ -426,49 +419,54 @@ def invoke(command):
                         child_processes.append(int(pids[0]))
                         parent_processes.append(int(pids[1]))
 
-                # the first item to in the PIDs we fetched corresponds run_get_output, which invokes the command using the
-                # shell, so in that case we need to use the parent's pid (which was the process we started)
-                started_commands_expected = parent_processes[0:1] + child_processes[1:]
+                # the first item to in the PIDs we fetched corresponds to run_get_output, which invokes the command using the
+                # shell, so in that case we need to use the parent's pid (i.e. the shell that we started)
+                started_commands = parent_processes[0:1] + child_processes[1:]
+
+                # wait for all the commands to start
+                running_commands = [[]]
```

Comment:
Out of curiosity, why are we using a 2D array here?

Reply:
all_commands_running() needs to change the value and would not be able to do it if it was scalar

Reply:
Ahh got it. I thought single arrays were modifiable outside their scope but apparently not, thanks for the clarification!